JavaScript/JavaScriptSDK/ajax/ajax.ts

```diff
@@ -177,27 +178,24 @@ module Microsoft.ApplicationInsights {
             });
         }
 
-        private onReadyStateChangePrefix(xhr: XMLHttpRequestInstrumented) {
-            switch (xhr.readyState) {
-                case 3:
-                    xhr.ajaxData.responseStartedTime = dateTime.Now();
-                    break;
-                case 4:
-                    this.collectResponseData(xhr);
-                    break;
-            }
-        }
-
-        private onReadyStateChangePostfix(xhr: XMLHttpRequestInstrumented) {
-            if (xhr.readyState === 4) {
-                this.onAjaxComplete(xhr);
-            }
-        }
-
         private onAjaxComplete(xhr: XMLHttpRequestInstrumented) {
-            try {
+            xhr.ajaxData.responseFinishedTime = dateTime.Now();
+            xhr.ajaxData.status = xhr.status;
                 xhr.ajaxData.CalculateMetrics();
 
+            if (xhr.ajaxData.ajaxTotalDuration < 0) {
```

Comment:
Tracing will give you requestURL, start time and stop time. Is it correct that start time and request URL are already sent with the ajax event itself? Am I missing something?